torch/lib/c10d/CMakeLists.txt

```diff
@@ -49,6 +49,14 @@ set(C10D_SRCS
   )
 
 add_library(c10d ${C10D_SRCS})
+target_compile_options(c10d PUBLIC
```

Comment:
Those probably should be private flags. You don't want to enforce warnings on library users.

Reply:
Good point. In our current use they are equivalent however.